[SDL 0173] - Read Generic Network Signal Data Clarification (#802)
smartdevicelink/sdl_evolution: proposals/0173-Read-Generic-Network-Signal-data.md

Here is a sample flow diagram for SDL Core <-> HMI request response model:
![SDL-HMI%20request%20response%20flow](../assets/proposals/0173-Read-Generic-Network-Signal-data/SDL-HMI%20request%20response%20flow.png)

#### API changes required for custom vehicle data:

- VehicleDataResult in MOBILE API will be extended with non mandatory parameter `oemCustomDataType` 

```
   <struct name="VehicleDataResult" since="2.0">
        <description>Individual published data request result</description>
        <param name="dataType" type="VehicleDataType" mandatory="true">
            <description>Defined published data element type.</description>
        </param>
        <param name="resultCode" type="VehicleDataResultCode" mandatory="true">
            <description>Published data result code.</description>
        </param>
+        <param name="oemCustomDataType" type="String" mandatory="false" since="6.0">

> +            <description>Type of requested oem specific parameter </description>

Reviewer: ```suggestion
+            <description>Type of requested OEM specific parameter </description>
```

+        </param>
    </struct>
```

`VehicleDataResult` in MOBILE API will be extended with non mandatory parameter `oemCustomDataType`

`VehicleDataType` will be extended with `VEHICLEDATA_OEM_CUSTOM_DATA` value in HMI and MOBILE APIs.
```
 <enum name="VehicleDataType" since="2.0">
        <description>Defines the data types that can be published and subscribed to.</description>
        <element name="VEHICLEDATA_GPS">
        ...
        <element name="VEHICLEDATA_CLOUDAPPVEHICLEID" since="5.1"/>
 +       <element name="VEHICLEDATA_OEM_CUSTOM_DATA" since="6.0"/>
 </enum>
```

> For oem specific custom vehicle data items, `oemCustomDataType` will contain type of OEM specific vehicle data (from schema), and `dataType` will be `VEHICLEDATA_OEM_CUSTOM_DATA`.

Reviewer: ```suggestion
For OEM specific custom vehicle data items, `oemCustomDataType` will contain type of OEM specific vehicle data (from schema), and `dataType` will be `VEHICLEDATA_OEM_CUSTOM_DATA`.
```

> For vehicle data items from RPCSpec, `oemCustomDataType` will be omitted, and `dataType` will contain appropriate data type from `VehicleDataType` enum.

Reviewer: ```suggestion
For vehicle data items from RPC Spec, `oemCustomDataType` will be omitted, and `dataType` will contain appropriate data type from `VehicleDataType` enum.
```
